### DISCUSSION TITLE
spring cloud gateway heapdump analysis

### DISCUSSION THREAD

**Discussion by xiaozaiyuji** [apache/skywalking#11073] [Q&A]
![img_v2_8fedd017-5463-4497-9141-55ab8fc02e0g](https://github.com/apache/skywalking/assets/4230320/3bf50f73-ccdc-4b27-9ede-c1bd6b729519)

why skywalking  use memory so much?

**Comment by wu-sheng** [comment]
With high noop span, this mean the trace context had been leaked.
It doesn't end expectedly.
But we can't tell why.
You are also able to see shadow dump in your agent logs as well.

**Reply by xiaozaiyuji** [reply]
I told the user to check the trace context and thead local.I think the connection is not release in time.This is a problem by use spring cloud gateway.If this is sovled,I will reply to you for the reason.
